pwndbg/aglib/shellcode.py

```diff
@@ -56,17 +57,18 @@ def exec_syscall(
     syscall_bin = pwnlib.asm.asm(syscall_asm)
 
     # Run the syscall and pass its return value onward to the caller.
-    return exec_shellcode(
+    async with exec_shellcode(
         ec,
         syscall_bin,
         restore_context=True,
-        capture=_get_syscall_return_value,
         disable_breakpoints=disable_breakpoints,
-    )
+    ):
+        return _get_syscall_return_value()
 
 
+@contextlib.asynccontextmanager
 async def exec_shellcode(
-    ec: ExecutionController, blob, restore_context=True, capture=None, disable_breakpoints=False
+    ec: ExecutionController, blob, restore_context=True, disable_breakpoints=False
```

Comment:
Why did we have `capture` in the first place? Why do we remove it now?

Reply:
Don't know why we used `capture`. In my opinion, using python `with` statement would be cleaner and more readable